observation_portal/accounts/serializers.py

```diff
@@ -105,3 +117,123 @@ class RevokeTokenResponseSerializer(serializers.Serializer):
+    """
+
+    # User model stuff
+    username = serializers.CharField(
+        max_length=150,
+        validators=[
+            validators.UniqueValidator(
+                queryset=User.objects.all(),
+                message="username already exists",
+                lookup="exact"
+            )
+        ]
+    )
+
+    # allow passing in the password, but if one is not provided generate one
+    # also don't return this in the response (write only)
+    password = serializers.CharField(
+        max_length=128, required=False, default=None, write_only=True
+    )
+    first_name = serializers.CharField(max_length=150)
+    last_name = serializers.CharField(max_length=150)
+    email = serializers.EmailField(
+        validators=[
+            validators.UniqueValidator(
+                queryset=User.objects.all(),
+                message="user with email already exists",
+                lookup="iexact"
+            )
+        ]
+    )
+
+    # Profile model stuff
+    institution = serializers.CharField(max_length=200)
+    title = serializers.CharField(max_length=200)
+    education_user = serializers.BooleanField(required=False, default=True)
+
+
+class BulkCreateUsersSerializer(serializers.Serializer):
+    users = serializers.ListField(child=CreateUserSerializer(), max_length=50)
+
+    def validate_users(self, data):
+        """Make sure usernames & emails are unique within the request payload"""
+        usernames = set()
+        emails = set()
+        for user in data:
+            username = user["username"]
+            if username in usernames:
+                raise serializers.ValidationError(f"username '{username}' provided multiple times")
+            usernames.add(username)
+
+            email = user["email"].lower()
+            if email in emails:
+                raise serializers.ValidationError(f"email '{email}' provided multiple times")
+            emails.add(email)
+
+        return data
+
+    def create_user(self, validated_data):
+        password = validated_data.pop("password", None)
+
+        if password is None:
+            password = User.objects.make_random_password(
+                length=12,
+                allowed_chars=string.ascii_letters + string.digits + "!@#$%^&*" * 3
+            )
+
+        user = User.objects.create_user(
+            username=validated_data.pop("username"),
+            password=password,
+            first_name=validated_data.pop("first_name"),
+            last_name=validated_data.pop("last_name"),
+            email=validated_data.pop("email"),
+            is_active=True,
+        )
+        Profile.objects.create(
+            user=user,
+            password_expiration=timezone.now(),
+            **validated_data
+        )
+
+        return password, user
+
+    def create_users(self, validated_data):
+        res = []
+        for user in validated_data:
+            res.append(self.create_user(user))
+        return res
+
+    def save(self):
```

Comment:
No particular reason...just wanted more control in the initial implementation. But I think I could do this in a `create` too.

Reply:
It was my understanding that save called either update or create based on if the item exists already. Since this endpoint is never updating, then overriding create makes more sense to me if that doesn't complicate things